core/templates/components/question-directives/questions-list/questions-list.component.ts

```diff
@@ -272,51 +272,69 @@ export class QuestionsListComponent implements OnInit, OnDestroy {
-            task: 'remove'
-          } as SkillLinkageModificationsArray
-        ]
-      ).then(() => {
-        this.questionsListService.resetPageNumber();
-        this.questionsListService.getQuestionSummariesAsync(
-          this.selectedSkillId, true, true);
-        this.alertsService.addSuccessMessage('Deleted Question');
-        this._removeArrayElement(questionId);
-      });
-    } else {
-      this.allSkillSummaries.forEach((summary) => {
-        if (summary.getDescription() === skillDescription) {
-          this.editableQuestionBackendApiService.editQuestionSkillLinksAsync(
-            questionId, [
-              {
-                id: summary.getId(),
-                task: 'remove'
-              } as SkillLinkageModificationsArray
-            ]
-          ).then(() => {
-            this.questionsListService.resetPageNumber();
-            this.questionsListService.getQuestionSummariesAsync(
-              this.selectedSkillId, true, true);
-            this.alertsService.addSuccessMessage('Deleted Question');
-            this._removeArrayElement(questionId);
-          });
+    let numberOfQuestions = this.questionSummariesForOneSkill.length;
+    await this.skillBackendApiService
+      .getTopicNamesWithGivenSkillAssignedForDiagnosticTest(
+        this.selectedSkillId).then((topicNames) => {
+        if ((topicNames.length > 0) && (
+          numberOfQuestions <=
+            AppConstants.MINIMUM_QUESTION_COUNT_FOR_A_DIAGNOSTIC_TEST_SKILL)) {
+          let alertMessge = (
+            'This skill is used as the only diagnostic test skill in the ' +
+            'following topics: ' + topicNames.join(', ') + '. Please remove ' +
+            'this skill from the diagnostic test of those topics first'
+          );
+          this.alertsService.addInfoMessage(alertMessge, 7000);
+          questionDeletionIsAllowed = false;
         }
       });
-    }
+    return questionDeletionIsAllowed;
+  }
+
+  editQuestionSkillLinks(questionId: string, skillId: string): void {
+    this.editableQuestionBackendApiService.editQuestionSkillLinksAsync(
+      questionId, [
+        {
+          id: skillId,
+          task: 'remove'
+        } as SkillLinkageModificationsArray
+      ]
+    ).then(() => {
+      this.questionsListService.resetPageNumber();
+      this.questionsListService.getQuestionSummariesAsync(
+        this.selectedSkillId, true, true);
+      this.alertsService.addSuccessMessage('Deleted Question');
+      this._removeArrayElement(questionId);
+    });
+  }
+
+  deleteQuestionFromSkill(
+      questionId: string, skillDescription: string): void {
+    this.isQuestionDeletionAllowed().then((response) => {
+      if (!response) {
+        return;
+      }
+
+      this.deletedQuestionIds.push(questionId);
+      if (this.allSkillSummaries.length === 0) {
+        this.editQuestionSkillLinks(questionId, this.selectedSkillId);
+      } else {
+        this.allSkillSummaries.forEach((summary) => {
```

Comment:
Can we use the [find method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find)?

Reply:
Updated